packages/material-ui/src/ButtonBase/TouchRipple.js

```diff
@@ -98,169 +98,194 @@ export const styles = theme => ({
+        };
+        // Delay the execution of the ripple effect.
+        startTimer.current = setTimeout(() => {
+          if (startTimerCommit.current) {
+            startTimerCommit.current();
+            startTimerCommit.current = null;
+          }
+        }, DELAY_RIPPLE); // We have to make a tradeoff with this value.
+      } else {
+        startCommit({ pulsate, rippleX, rippleY, rippleSize, cb });
+      }
+    },
+    [centerProp, startCommit],
+  );
 
-    this.setState(
-      state => ({
-        nextKey: state.nextKey + 1,
-        ripples: [
-          ...state.ripples,
-          <Ripple
-            key={state.nextKey}
-            classes={this.props.classes}
-            timeout={DURATION}
-            pulsate={pulsate}
-            rippleX={rippleX}
-            rippleY={rippleY}
-            rippleSize={rippleSize}
-          />,
-        ],
-      }),
-      cb,
-    );
-  };
+  const pulsate = React.useCallback(() => {
+    start({}, { pulsate: true });
+  }, [start]);
 
-  stop = (event, cb) => {
-    clearTimeout(this.startTimer);
+  const stop = React.useCallback((event, cb) => {
+    clearTimeout(startTimer.current);
 
     // The touch interaction occurs too quickly.
     // We still want to show ripple effect.
-    if (event.type === 'touchend' && this.startTimerCommit) {
+    if (event.type === 'touchend' && startTimerCommit.current) {
       event.persist();
-      this.startTimerCommit();
-      this.startTimerCommit = null;
-      this.startTimer = setTimeout(() => {
-        this.stop(event, cb);
+      startTimerCommit.current();
+      startTimerCommit.current = null;
+      startTimer.current = setTimeout(() => {
+        stop(event, cb);
       });
       return;
     }
 
-    this.startTimerCommit = null;
+    startTimerCommit.current = null;
 
-    this.setState(({ ripples }) => {
-      if (ripples && ripples.length) {
-        return { ripples: ripples.slice(1) };
+    setRipples(oldRipples => {
+      if (oldRipples.length > 0) {
+        return oldRipples.slice(1);
       }
-      return null;
-    }, cb);
-  };
+      return oldRipples;
+    });
+    rippleCallback.current = cb;
+  }, []);
+
+  React.useImperativeHandle(
+    ref,
+    () => ({
+      pulsate,
+      start,
+      stop,
+    }),
+    [pulsate, start, stop],
+  );
 
-  render() {
-    const { center, classes, className, ...other } = this.props;
+  return (
+    <span className={clsx(classes.root, className)} ref={container} {...other}>
+      <TransitionGroup component={null} exit>
+        {ripples}
+      </TransitionGroup>
+    </span>
+  );
+});
 
-    return (
-      <span className={clsx(classes.root, className)} ref={this.container} {...other}>
-        <TransitionGroup component={null} exit>
-          {this.state.ripples}
-        </TransitionGroup>
-      </span>
-    );
-  }
+// TODO cleanup after https://github.com/reactjs/react-docgen/pull/378 is released
+function withMuiName(Component) {
```

Comment:
One line above is a comment. Could you please elaborate a bit why you think this is resolved?

Reply:
The component is no longer documented. I have tried without this HOC locally. It seems alright.

Reply:
But the component was public previously. We can't just remove it in a minor release.

Reply:
Is this a breaking change? What's the definition of a public component? If it's behing exported in the barrel index, it was private.

Reply:
The props were documented.

Reply:
Given that the prop is documented in the ButtonBase, that we have an API page for it, that the style name is not prefixed by mui. I agree, the component should be considered public.

What about removing the documentation of a public API we want to deprecate in the next few months?

Reply:
Hmm, I’m happy to add it back in and hide the ref stuff or keep it in. I think it should be public still since it’s public at the moment. If we don’t want to encourage the usage of it separate to the ButtonBase we should hide the ref details but I think there isn’t any harm documenting everything.

Reply:
I'm happy either way too. I think that, as long as, we make TouchRipple private in v5, we will be fine.